s3/tests/unit.tftest.hcl

```diff
@@ -185,7 +185,41 @@ run "aws_s3_bucket_lifecycle_configuration_unit_test" {
 
   assert {
     condition     = aws_s3_bucket_lifecycle_configuration.lifecycle-configuration[0].rule[0].expiration[0].days == 90
-    error_message = "Should be: /foo"
+    error_message = "Should be: 90"
+  }
+
+  assert {
+    condition     = aws_s3_bucket_lifecycle_configuration.lifecycle-configuration[0].rule[0].abort_incomplete_multipart_upload[0].days_after_initiation == 7
+    error_message = "Should be: 7"
+  }
+}
+
+run "aws_s3_bucket_lifecycle_configuration_no_prefix_unit_test" {
+  command = plan
+
+  variables {
+    config = {
+      "bucket_name"     = "dbt-terraform-test-s3-module",
+      "type"            = "string",
+      "lifecycle_rules" = [{ "expiration_days" = 90, "enabled" = true }],
+      "objects"         = [],
+    }
+  }
+
+  ### Test aws_s3_bucket_lifecycle_configuration resource when no prefix is used ###
+  assert {
+    condition     = aws_s3_bucket_lifecycle_configuration.lifecycle-configuration[0].rule[0].filter[0] != null
+    error_message = "Should be: {}"
```

Comment:
Why is `aws_s3_bucket_lifecycle_configuration.lifecycle-configuration[0].rule[0].filter[0].prefix` a string when there's a filter and an empty map when there's no filter?

Reply:
the `filter` should be an empty map `{}` if there's no prefix.
If there's a prefix, the `filter` should have a value in its `prefix` field.

Reply:
The reason for the empty map in the filter field is to allow a rule which applies to **all objects** in the bucket `{}`, as opposed to a set of rules where rule 1 and 2 might include a prefix, and rule 4 has a prefix with an empty string to catch the rest.  If a single rule exists with prefix "", then it would be equivalent to {}